src/components/BrowserCell/BrowserCell.react.js

```diff
@@ -23,12 +23,149 @@ export default class BrowserCell extends Component {
     this.cellRef = React.createRef();
     this.copyableValue = undefined;
     this.state = {
-      showTooltip: false
-    }
-
+      showTooltip: false,
+      content: null,
+      classes: []
+    };
   }
 
-  componentDidUpdate(prevProps) {
+  async componentDidUpdate(prevProps) {
+    if ( this.props.value !== prevProps.value ) {
+      let content = this.props.value;
+      let isNewRow = this.props.row < 0;
+      this.copyableValue = content;
+      let classes = [styles.cell, unselectable];
+      if (this.props.hidden) {
+        content = this.props.value !== undefined || !isNewRow ? '(hidden)' : this.props.isRequired ? '(required)' : '(undefined)';
+        classes.push(styles.empty);
+      } else if (this.props.value === undefined) {
+        if (this.props.type === 'ACL') {
+          this.copyableValue = content = 'Public Read + Write';
+        } else {
+          this.copyableValue = content = '(undefined)';
+          classes.push(styles.empty);
+        }
+        content = isNewRow && this.props.isRequired && this.props.value === undefined ? '(required)' : content;
+      } else if (this.props.value === null) {
+        this.copyableValue = content = '(null)';
+        classes.push(styles.empty);
+      } else if (this.props.value === '') {
+        content = <span>&nbsp;</span>;
+        classes.push(styles.empty);
+      } else if (this.props.type === 'Pointer') {
+        const defaultPointerKey = await localStorage.getItem(this.props.value.className) || 'objectId';
```

Comment:
I assume there is currently no caching mechanism, correct? So if the same pointer is used with the same objectId in multiple rows, there will be a DB request for each cell? I would not make this is a requirement for this PR, but I think that would be one of the possible improvements we could mention in the top post of this PR, so people are aware of this caveat.

Reply:
Yes, it'll perform a DB operation for each row. I'll mention it on the PR.